example/spec/serial/multiQueue/ParsePdrBatchQueueSpec.js

```diff
@@ -0,0 +1,280 @@
+        'ParsePdr',
+        testCollection1,
+        testProvider,
+        parsePdrPayload,
+        { queueBatchSize: 4 }
+      )).executionArn;
+
+      const lambdaStep = new LambdaStep();
+
+      await pWaitFor(
+        async () => {
+          const { status } = await getExecution({
+            prefix: stackName,
+            arn: parsePdrExecutionArn,
+          });
+
+          return status === 'completed';
+        },
+        { interval: 2000, timeout: 60000 }
+      );
+
+      parsePdrOutput = await lambdaStep.getStepOutput(
+        parsePdrExecutionArn,
+        'ParsePdr'
+      );
+
+      queueGranulesOutput = await lambdaStep.getStepOutput(
+        parsePdrExecutionArn,
+        'QueueGranules'
+      );
+    } catch (error) {
+      beforeAllError = error;
+    }
+  });
+
+  afterAll(async () => {
+    await Promise.all(testGranuleIds.map(
+      (granuleId) => waitForGranuleAndDelete(stackName, granuleId, 'completed')
+    ));
+    await deletePdr({ prefix: stackName, pdrName });
+    await Promise.all(queueGranulesOutput.payload.running.map(
+      (executionArn) => deleteExecution({ prefix: stackName, executionArn })
+    ));
+    await deleteExecution({ prefix: stackName, executionArn: parsePdrExecutionArn });
+    await Promise.all([
+      deleteFolder(bucket, testDataPath),
+      deleteFolder(nodeNameBucket, testDataPath),
+      deleteCollection({
+        prefix: stackName,
+        collectionName: testCollection1.name,
+        collectionVersion: testCollection1.version,
+      }),
+      deleteCollection({
+        prefix: stackName,
+        collectionName: testCollection2.name,
+        collectionVersion: testCollection2.version,
+      }),
+      deleteProvider({
+        prefix: stackName,
+        provider: testProvider.id,
+      }),
+      deleteProvider({
+        prefix: stackName,
+        provider: nodeNameProvider.id,
+      }),
+    ]);
+  });
+
+  it('yields the expected output of granules in the payload', () => {
+    if (beforeAllError) fail(beforeAllError);
+    else {
+      expect(parsePdrOutput.payload.granulesCount).toEqual(testGranuleIds.length);
+      expect(parsePdrOutput.payload.granules.map((g) => g.granuleId)).toEqual(testGranuleIds);
+    }
+  });
+
+  describe('and the queue-granules task', () => {
+    it('queues granules into the expected number of workflows', () => {
+      if (beforeAllError) fail(beforeAllError);
+      else expect(queueGranulesOutput.payload.running.length).toEqual(3);
+    });
+
+    it('creates queued workflow inputs in expected batches', async () => {
+      if (beforeAllError) fail(beforeAllError);
+      else {
+        const executionArns = queueGranulesOutput.payload.running;
+        const inputs = await Promise.all(executionArns.map(getExecutionInputObject));
+        const granuleIds = inputs.map((input) => input.payload.granules.map((granule) => granule.granuleId)).flat();
+        expect(granuleIds.sort()).toEqual(testGranuleIds);
```

Comment:
Do we want/need to test that each workflow contains the right number of granules?